docs/readmes/contributing/contribute_commit_msg.md

```diff
@@ -0,0 +1,85 @@
+# Improve git commit messages
+This document is about recommended git commit style.  Every project generally
+has conventions for writing commit message subject line.  Rather than defining
+our own convention, magma can make use of commit subject structure
+according to conventionalcommits.org.
+
+#$ tldr
+Commit message can follow the template defined in magma github.
+This document covers commit subject line structure and
+validation APP.
+```
+<type>[optional scope]: <description>
+```
+for example:
+```
+fix(AGW): fix pyroute2 dependency
+```
+
+## Advantage of semantic commits.
+1. This state intention of the commit msg in concise manner in PR title.
+2. It can be used for automatically compiling change log for magma release
+   documentation.
+3. Analytics related to number of fixes vs feature commits in each release.
+4. This would force developer to rethink when they mix different type of
+   changes in same PR.
+
+## type
```

Comment:
For type and scope it might be better to link to the config file so that the doc stays up to date.

Reply:
I am not sure how to do it, can you point me to example?

Reply:
I didn't mean anything fancy. I just meant adding a line like: `for an up-to-date list of types we support see <INSERT LINK TO https://github.com/magma/magma/blob/master/.github/semantic.yml>`

Just so that people know where to look. (I know you mention the file in the doc somewhere else, but people might not be careful readers :) )